codegen-test/model/rest-json-extras.smithy

```diff
@@ -302,7 +307,8 @@ operation CaseInsensitiveErrorOperation {
         code: 500,
         body: "{\"Message\": \"hello\"}",
         headers: { "X-Amzn-Errortype": "CaseInsensitiveError" },
-        params: { message: "hello" }
+        params: { message: "hello" },
+        appliesTo: "client",
```

Comment:
I would expand the documentation to justify why this is a client-only test; took me a while to figure out why. It turns out that smithy-rs clients should be able to parse error messages if the JSON keys them as `message`, `Message`, or `errorMessage`. The relevant parsing code is [here](https://github.com/awslabs/smithy-rs/blob/6c21fb0eb377c7120a8179f4537ba99a4b50ba96/rust-runtime/inlineable/src/json_errors.rs#L51-L51).

Reply:
Added a comment with the link